### ISSUE TITLE
✨ [REQUEST] Birdnet-Go

### ISSUE THREAD

**Issue by matthew73210** [alexbelgium/hassio-addons#1385]
### Checked

- [X] If a new addon, I have checked on  Google that such as addon doesn't already exists
- [X] If a new feature, I have searched in the discussions & issues that it was not already covered

### Which addon?

Birdnet-go

### Is your feature request related to a problem? Please describe

No it is not.

### Describe the solution you'd like

Hey! Would it be possible to have the addon autmoaticlly map (instead of adding it to config) the share patition so that clips and webui.log are availble at the get-go?
Cheers

**Comment by matthew73210** [comment]
Sometimes birdnet-go stops analysing, would there be a way to monitor the logs and if it does say not get a reading for 10 mins restart the container?

**Comment by alexbelgium** [comment]
> Sometimes birdnet-go stops analysing, would there be a way to monitor the logs and if it does say not get a reading for 10 mins restart the container?

Hi, I've done that for birdnet-pi but it was much easier as based on services. I'll have to look how to do it for the go version

**Comment by alexbelgium** [comment]
For your main question you can already do it by mapping a drive and using the relative path (such as /mnt/NAS/Clips instead of the default "Clips" which is relative to the home folder)

I could do it by default i'll have a look

**Comment by deadrubberboy** [comment]
Hi Alex! What are the differences between Pi and Go? Can you still join bird weather with go? As you know I've been testing and playing with Pi and enjoying it. I am curious about the lower system load of Go though.

Also - is it possible to run your Bridget-Pi as a stand alone docker container outside of Home Assistant?

**Comment by alexbelgium** [comment]
Hi, here is my view from what I know :
- Birdnet-pi : historical app abandoned by its owner but with a very active fork to keep it alive, lots of functions, runs on x64 (such as rpi4, x86). With it running, average cpu usage is <20% and analysis of a 15s segment lasts 2.5 seconds.
- Birdnet-go : new app under heavy development, minimalistic UI, more resources efficient (for example runs on rpi3)

Both can join birdweather!

I'm all for the dev of a new system but as I user I'm selfishly using birdnet-pi as it is much more mature with features of analysis, deletion, species change... Also it is based on php/py/sh which means I can contribute as a contrary of birdnet-go that is in go which I know nothing about.

Currently you can't run it as a standalone container but I think another user was developing one...

**Comment by deadrubberboy** [comment]
Thanks! Considering I don’t use the change species.  Would I lose anything going to your Go Addon? Is species detection just as good on Go?

**Comment by alexbelgium** [comment]
I haven't really tested it! What you could do is run both add-ons in parallel with the same rtsp feed. I've ran 2 birdnet pi add-ons in parallel without any impact on performance of the rpi4!

**Comment by deadrubberboy** [comment]
Got it. What about Size? That's really the problem I'm running into more. My HA VM was 20gigs and now it's 60 and I'm trying to figure it why.

**Comment by alexbelgium** [comment]
I'm not sure about size ; birdnet-pi is based on "ghcr.io/linuxserver/baseimage-debian:arm64v8-bookworm" and birdnet-go on "ghcr.io/tphakala/birdnet-go:dev"

Try running `ha su repair` it will remove all dangling images that take a lot of place

**Comment by deadrubberboy** [comment]
Thanks! I got birdnet-go running. Is there a way to tell it to purge recordings daily or anything? Also, the log and stats both say "nothing here yet" even though it is detecting.  Am I just being impatient? 🙂

**Comment by deadrubberboy** [comment]
update: found the yaml that I can set logs to "true" - will that also be stats?

**Comment by alexbelgium** [comment]
> Thanks! I got birdnet-go running. Is there a way to tell it to purge recordings daily or anything? Also, the log and stats both say "nothing here yet" even though it is detecting. Am I just being impatient? 🙂

I'm not sure I think there is a purge but without much granularity - that's why I'm using birdnet-pi (I'm also developing a feature to keep only the best & latest recordings to avoid having too much data)

For the log and stat I think it's because it is placeholder pages

**Comment by deadrubberboy** [comment]
I started using the birdnet go.  I uninstalled birdnet pi and my home assistant full backup size went from almost 8gigs down to 400k. That seems unrealistic.  Anything that would make that true? I fear my backups are somehow incomplete now?

**Comment by alexbelgium** [comment]
In a backup you can look in the zip which files take a lot of place. It could be simply the mp3 files from the saved detections, and birdnet go will fill the same in some time

**Comment by deadrubberboy** [comment]
They don’t purge at some point?

**Comment by alexbelgium** [comment]
When the disk is at 95% capacity :) I'm writing a feature in birdnet pi to define a maximum of recordings to keep per species. For example i had 47k mp3 recordings of blackbird. The objective will be to keep the last 7 days, plus x (fire example 1000) recordings sorted by highest confidence

**Comment by alexbelgium** [comment]
Hi, was it working now or is it still an open request? thanks!

**Comment by deadrubberboy** [comment]
Safe to close. Thanks!
